windows/src/desktop/kmshell/util/Keyman.System.KeymanStartTask.pas

```diff
@@ -0,0 +1,133 @@
+unit Keyman.System.KeymanStartTask;
+
+interface
+
+type
+  /// <summary>Create or recreate the automatic task to start Keyman if a TIP
+  /// is selected based on an event in the Event Log</summary>
+  TKeymanStartTask = class
+  private
+  public
+    class procedure RecreateTask;
+    class procedure DeleteTask;
+    class procedure CreateTask;
+  end;
+
+implementation
+
+uses
+  System.Variants,
+  Winapi.ActiveX,
+  Winapi.Windows,
+
+  DebugPaths,
+  KeymanPaths,
+  KeymanVersion,
+  RegistryKeys,
+  TaskScheduler_TLB;
+
+{ TKeymanStartTask }
+
+const
+  CTaskName = 'Start On Demand';
+  CTaskDescription = 'Starts Keyman on demand when a Keyman input method is selected';
+  CTaskAuthor = 'SIL International';
+  CTaskFolderName = 'Keyman';
+  CTaskTrigger =
+    '<QueryList>'+
+      '<Query Id="0" Path="Application">'+
+        '<Select Path="Application">*[System[Provider[@Name=''Keyman''] and EventID=256]]</Select>'+
```

Comment:
> it creates a task that is launched whenever a specific Keyman event with id 256 is added to the Event Log

Of all the id's, why `256`?

Reply:
`0x100` was the event id used in the example. It's just a number, not important.